src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java

```diff
@@ -521,6 +521,9 @@ private static void validateCheckSection(ModuleFactory moduleFactory, String fil
                     fileName + " section '" + sectionName + "' should be in order");
 
             switch (subSectionPos) {
+                case 0:
+                    validateDescriptionSection(fileName, sectionName, subSection);
```

Comment:
In line 495
```
if (subSectionPos == 0 && "p".equals(subSection.getNodeName())) {
                validateSinceDescriptionSection(fileName, sectionName, subSection);
                continue;
            }
```
It looks like fragment above can be moved to `validateDescriptionSection` method

Reply:
No, this is a special case for the `<p>` section which has no `name` attribute. By moving this code to the `validateDescriptionSection` method a NPE occurs on the next line.

Reply:
This test is a bit complicated. Perhaps, this test should be split into two: one to validate the order of the sections, another to check the content of each section.